sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueSenderAsyncClient.java

```diff
@@ -0,0 +1,374 @@
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Void> send(Iterable<Message> messages) {
+        Objects.requireNonNull(messages, "'messages' cannot be null.");
+        return send(Flux.fromIterable(messages));
+    }
+
+    /**
+     * Sends a set of messages to the associated Event Hub using a batched approach. If the size of messages exceed the
+     * maximum size of a single batch, an exception will be triggered and the send will fail. By default, the message
+     * size is the max amount allowed on the link.
+     * @param messages Events to send to the service.
+     *
+     * @return A {@link Mono} that completes when all messages are pushed to the service.
+     */
+    public Mono<Void> send(Flux<Message> messages) {
+        Objects.requireNonNull(messages, "'messages' cannot be null.");
+
+        return sendInternal(messages);
+    }
+
+    private Mono<Void> sendInternal(Flux<Message> messages) {
+        return getSendLink()
+            .flatMap(link -> link.getLinkSize()
+                .flatMap(size -> {
+                    final int batchSize = size > 0 ? size : MAX_MESSAGE_LENGTH_BYTES;
+                    final CreateBatchOptions batchOptions = new CreateBatchOptions()
+                        .setMaximumSizeInBytes(batchSize);
+                    return messages.collect(new AmqpMessageCollector(batchOptions, 1,
+                        link::getErrorContext, tracerProvider));
+                })
+                .flatMap(list -> sendInternalBatch(Flux.fromIterable(list))));
+    }
+
+    private Mono<Void> sendInternalBatch(Flux<MessageBatch> eventBatches) {
+        return eventBatches
+            .flatMap(this::send)
+            .then()
+            .doOnError(error -> {
+                logger.error("Error sending batch.", error);
+            });
+    }
+
+
+    /**
+     *
+     * @param message to be scheduled
+     * @param scheduledEnqueueTimeUt Time of the enqueue.
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Long> schedule(Message message, Instant scheduledEnqueueTimeUt) {
+        //TODO(feature-to-implement)
+        return null;
+    }
+
+    /**
+     *
+     * @param sequenceNumber for the message to cancel.
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Void> cancelScheduledMessage(long sequenceNumber) {
+        //TODO(feature-to-implement)
+        return null;
+    }
+
+    /**
+     *
+     * @return The name of  the queue.
+     */
+    public String getQueueName() {
+        return this.queueName;
+    }
+
+    /**
+     *
+     * @param batch of messages which allows client to send maximum allowed size for a batch of messages.
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Void> send(MessageBatch batch) {
```

Comment:
This should be package-private. MessageBatch is not accessible outside of this package.